frontend/viewer/src/lib/entry-editor/object-editors/AddSenseFab.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+  import {fieldName} from '$lib/i18n';
+  import {useCurrentView} from '$lib/services/view-service';
+  import {mdiPlus} from '@mdi/js';
+  import throttle from 'just-throttle';
+  import {onMount} from 'svelte';
+  import {Button} from 'svelte-ux';
+
+  const currentView = useCurrentView();
+
+  let shrinkFab = false;
+
+  const handleScroll = throttle((element: Element | null) => {
+    if (!element) return;
+    if (element.scrollTop == 0) {
+      shrinkFab = false;
+      return;
+    }
+    const atEndOfScroll = element.clientHeight + element.scrollTop >= element.scrollHeight - 10;
+    shrinkFab = !atEndOfScroll;
```

Comment:
help me understand what's going on here? what's the goal of handle scroll?

Reply:
Here's how it behaves:

https://github.com/user-attachments/assets/c3addcf9-d130-4a2c-8cde-2f6d650d7d0a